## 1. The oldest business in the world
<p><img src="https://assets.datacamp.com/production/repositories/5851/datasets/7dded924c6dc418d4a828f2f4daba99953c27a5a/400px-Eingang_zum_St._Peter_Stiftskeller.jpg" alt="The entrance to St. Peter Stiftskeller, a restaurant in Saltzburg, Austria. The sign over the entrance shows &quot;803&quot; - the year the business opened."></p>
<p><em>Image: St. Peter Stiftskeller, founded 803. Credit: <a href="https://commons.wikimedia.org/wiki/File:Eingang_zum_St._Peter_Stiftskeller.jpg">Pakeha</a>.</em></p>
<p>An important part of business is planning for the future and ensuring that the company survives changing market conditions. Some businesses do this really well and last for hundreds of years.</p>
<p>BusinessFinancing.co.uk <a href="https://businessfinancing.co.uk/the-oldest-company-in-almost-every-country">researched</a> the oldest company that is still in business in (almost) every country and compiled the results into a dataset. In this project, you'll explore that dataset to see what they found.</p>
<p>The database contains three tables.</p>
<h3 id="categories"><code>categories</code></h3>
<table>
<thead>
<tr>
<th style="text-align:left;">column</th>
<th>type</th>
<th>meaning</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;"><code>category_code</code></td>
<td>varchar</td>
<td>Code for the category of the business.</td>
</tr>
<tr>
<td style="text-align:left;"><code>category</code></td>
<td>varchar</td>
<td>Description of the business category.</td>
</tr>
</tbody>
</table>
<h3 id="countries"><code>countries</code></h3>
<table>
<thead>
<tr>
<th style="text-align:left;">column</th>
<th>type</th>
<th>meaning</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;"><code>country_code</code></td>
<td>varchar</td>
<td>ISO 3166-1 3-letter country code.</td>
</tr>
<tr>
<td style="text-align:left;"><code>country</code></td>
<td>varchar</td>
<td>Name of the country.</td>
</tr>
<tr>
<td style="text-align:left;"><code>continent</code></td>
<td>varchar</td>
<td>Name of the continent that the country exists in.</td>
</tr>
</tbody>
</table>
<h3 id="businesses"><code>businesses</code></h3>
<table>
<thead>
<tr>
<th style="text-align:left;">column</th>
<th>type</th>
<th>meaning</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;"><code>business</code></td>
<td>varchar</td>
<td>Name of the business.</td>
</tr>
<tr>
<td style="text-align:left;"><code>year_founded</code></td>
<td>int</td>
<td>Year the business was founded.</td>
</tr>
<tr>
<td style="text-align:left;"><code>category_code</code></td>
<td>varchar</td>
<td>Code for the category of the business.</td>
</tr>
<tr>
<td style="text-align:left;"><code>country_code</code></td>
<td>char</td>
<td>ISO 3166-1 3-letter country code.</td>
</tr>
</tbody>
</table>
<p>Let's begin by looking at the range of the founding years throughout the world.</p>

In [2]:
%%sql 
postgresql:///oldestbusinesses
    select max(year_founded) as new, min(year_founded) as min
    from businesses
 
-- Select the oldest and newest founding years from the businesses table


1 rows affected.


[(1999, 578)]

## 2. How many businesses were founded before 1000?
<p>Wow! That's a lot of variation between countries. In one country, the oldest business was only founded in 1999. By contrast, the oldest business in the world was founded back in 578. That's pretty incredible that a business has survived for more than a millennium.</p>
<p>I wonder how many other businesses there are like that.</p>

In [3]:
%%sql

-- Get the count of rows in businesses where the founding year was before 1000
select count(*)
from businesses
where year_founded < 1000;

 * postgresql:///oldestbusinesses
1 rows affected.


[(6,)]

## 3. Which businesses were founded before 1000?
<p>Having a count is all very well, but I'd like more detail. Which businesses have been around for more than a millennium?</p>

In [4]:
%%sql

-- Select all columns from businesses where the founding year was before 1000
-- Arrange the results from oldest to newest
select * from businesses
where year_founded < 1000
order by year_founded

 * postgresql:///oldestbusinesses
6 rows affected.


[('Kongō Gumi', 578, 'CAT6', 'JPN'),
 ('St. Peter Stifts Kulinarium', 803, 'CAT4', 'AUT'),
 ('Staffelter\xa0Hof Winery', 862, 'CAT9', 'DEU'),
 ('Monnaie de Paris\xa0', 864, 'CAT12', 'FRA'),
 ('The Royal Mint', 886, 'CAT12', 'GBR'),
 ("Sean's Bar", 900, 'CAT4', 'IRL')]

## 4. Exploring the categories
<p>Now we know that the oldest, continuously operating company in the world is called Kongō Gumi. But was does that company do? The category codes in the <code>businesses</code> table aren't very helpful: the descriptions of the categories are stored in the <code>categories</code> table.</p>
<p>This is a common problem: for data storage, it's better to keep different types of data in different tables, but for analysis, you want all the data in one place. To solve this, you'll have to join the two tables together.</p>

In [5]:
%%sql
select * from businesses

 * postgresql:///oldestbusinesses
163 rows affected.


[('Hamoud Boualem', 1878, 'CAT11', 'DZA'),
 ('Communauté Électrique du Bénin', 1968, 'CAT10', 'BEN'),
 ('Botswana Meat Commission', 1965, 'CAT1', 'BWA'),
 ('Air Burkina', 1967, 'CAT2', 'BFA'),
 ('Brarudi', 1955, 'CAT9', 'BDI'),
 ('Cameroon Development Corporation', 1947, 'CAT1', 'CMR'),
 ('Correios de Cabo Verde', 1849, 'CAT16', 'CPV'),
 ('Banque Internationale pour la Centrafrique', 1946, 'CAT3', 'CAF'),
 ('Cotontchad', 1971, 'CAT1', 'TCD'),
 ('Central Bank of the Comoros', 1981, 'CAT3', 'COM'),
 ('Société nationale des Chemins de fer du Congo', 1889, 'CAT2', 'COD'),
 ('Development Bank of the Central African States', 1975, 'CAT3', 'COG'),
 ('La Poste', 1945, 'CAT16', 'CIV'),
 ('Ethio-Djibouti Railways', 1901, 'CAT2', 'DJI'),
 ('Egyptian National Railways', 1854, 'CAT2', 'EGY'),
 ('Guinea Ecuatorial Airlines', 1996, 'CAT2', 'GNQ'),
 ('Asmara Brewery', 1939, 'CAT9', 'ERI'),
 ('National Bank of Ethiopia', 1906, 'CAT3', 'ETH'),
 ('BGFIBank Group', 1971, 'CAT3', 'GAB'),
 ('Halco Mining', 1962, 'CAT15', 'GIN'),
 ('Correios da Guiné-Bissau', 1973, 'CAT16', 'GNB'),
 ('KCB Group Limited', 1896, 'CAT3', 'KEN'),
 ('Central Bank of Lesotho', 1978, 'CAT3', 'LSO'),
 ('National Port Authority', 1921, 'CAT2', 'LBR'),
 ('Umma Bank', 1907, 'CAT3', 'LBY'),
 ('Air Madagascar', 1962, 'CAT2', 'MDG'),
 ('Malawi Broadcasting Corporation', 1964, 'CAT13', 'MWI'),
 ('Office de Radiodiffusion-Télévision du Mali', 1957, 'CAT13', 'MLI'),
 ('Central Bank of Mauritania', 1973, 'CAT3', 'MRT'),
 ('Mauritius Post', 1772, 'CAT16', 'MUS'),
 ('Attijariwafa Bank', 1904, 'CAT3', 'MAR'),
 ('Beira Railroad Corporation', 1892, 'CAT2', 'MOZ'),
 ('NamPost', 1814, 'CAT16', 'NAM'),
 ('Office of Radio and Television of Niger', 1960, 'CAT13', 'NER'),
 ('First Bank of Nigeria', 1894, 'CAT3', 'NGA'),
 ('National Post Office', 1922, 'CAT16', 'RWA'),
 ('Central Bank of São Tomé and Príncipe', 1975, 'CAT3', 'STP'),
 ('Dakar–Niger Railway', 1924, 'CAT2', 'SEN'),
 ('Air Seychelles', 1977, 'CAT2', 'SYC'),
 ('Rokel Commercial Bank', 1917, 'CAT3', 'SLE'),
 ('Radio Mogadishu', 1943, 'CAT13', 'SOM'),
 ('Premier FMCG', 1820, 'CAT12', 'ZAF'),
 ('Ivory Bank', 1994, 'CAT3', 'SSD'),
 ('Bank of Khartoum', 1913, 'CAT3', 'SDN'),
 ('Swazi Rail', 1963, 'CAT2', 'SWZ'),
 ('Tanzania Breweries Limited', 1933, 'CAT9', 'TZA'),
 ('La Poste du Togo', 1883, 'CAT16', 'TGO'),
 ('La Poste Tunisienne', 1847, 'CAT16', 'TUN'),
 ('Stanbic Bank Uganda Limited', 1906, 'CAT3', 'UGA'),
 ('ZamPost', 1896, 'CAT16', 'ZMB'),
 ('Standard Chartered Zimbabwe', 1892, 'CAT3', 'ZWE'),
 ('Spinzar Cotton Company', 1930, 'CAT1', 'AFG'),
 ('Azerbaijan Caspian Shipping Company', 1858, 'CAT2', 'AZE'),
 ('BMMI', 1883, 'CAT17', 'BHR'),
 ('M. M. Ispahani Limited', 1820, 'CAT11', 'BGD'),
 ('Tashi Group', 1959, 'CAT2', 'BTN'),
 ('Hua Ho Department Store', 1947, 'CAT17', 'BRN'),
 ('National Bank of Cambodia', 1954, 'CAT3', 'KHM'),
 ("Ma Yu Ching's Bucket Chicken House", 1153, 'CAT4', 'CHN'),
 ('Bank of Georgia', 1903, 'CAT3', 'GEO'),
 ('Wadia Group', 1736, 'CAT12', 'IND'),
 ('Pindad', 1808, 'CAT8', 'IDN'),
 ('North Oil Company', 1928, 'CAT10', 'IRQ'),
 ('Café Abu Salem', 1914, 'CAT4', 'ISR'),
 ('Kongō Gumi', 578, 'CAT6', 'JPN'),
 ('Arab Bank', 1930, 'CAT3', 'JOR'),
 ('Bogatyr Access Komir', 1913, 'CAT15', 'KAZ'),
 ('M.H. Alshaya Co.', 1890, 'CAT17', 'KWT'),
 ('Electricite du Laos', 1959, 'CAT10', 'LAO'),
 ('Bank Audi', 1830, 'CAT3', 'LBN'),
 ('Pos Malaysia', 1800, 'CAT16', 'MYS'),
 ('Mongolian National Broadcaster', 1931, 'CAT13', 'MNG'),
 ('Myanmar National Airlines', 1948, 'CAT2', 'MMR'),
 ('Nepal Bank Limited', 1937, 'CAT3', 'NPL'),
 ("Kim Chong-t'ae Electric Locomotive Works", 1945, 'CAT2', 'PRK'),
 ('Petroleum Development Oman', 1937, 'CAT10', 'OMN'),
 ('House of Habib', 1841, 'CAT5', 'PAK'),
 ('Destileria Limtuaco', 1853, 'CAT9', 'PHL'),
 ('Salam International Investment Limited', 1952, 'CAT5', 'QAT'),
 ('Petrodvorets Watch Factory', 1721, 'CAT7', 'RUS'),
 ('House of Alireza', 1845, 'CAT6', 'SAU'),
 ('Singapore Post', 1819, 'CAT1

In [7]:
%%sql
select * from categories

 * postgresql:///oldestbusinesses
19 rows affected.


[('CAT1', 'Agriculture'),
 ('CAT2', 'Aviation & Transport'),
 ('CAT3', 'Banking & Finance'),
 ('CAT4', 'Cafés, Restaurants & Bars'),
 ('CAT5', 'Conglomerate'),
 ('CAT6', 'Construction'),
 ('CAT7', 'Consumer Goods'),
 ('CAT8', 'Defense'),
 ('CAT9', 'Distillers, Vintners, & Breweries'),
 ('CAT10', 'Energy'),
 ('CAT11', 'Food & Beverages'),
 ('CAT12', 'Manufacturing & Production'),
 ('CAT13', 'Media'),
 ('CAT14', 'Medical'),
 ('CAT15', 'Mining'),
 ('CAT16', 'Postal Service'),
 ('CAT17', 'Retail'),
 ('CAT18', 'Telecommunications'),
 ('CAT19', 'Tourism & Hotels')]

In [13]:
%%sql

-- Select business name, founding year, and country code from businesses; and category from categories
-- where the founding year was before 1000, arranged from oldest to newest
select b.business, b.year_founded, b.country_code, b.category_code, c.category
from businesses as b
join categories as c
on b.category_code = c.category_code
where b.year_founded < 1000
order by b.year_founded

 * postgresql:///oldestbusinesses
6 rows affected.


[('Kongō Gumi', 578, 'JPN', 'CAT6', 'Construction'),
 ('St. Peter Stifts Kulinarium', 803, 'AUT', 'CAT4', 'Cafés, Restaurants & Bars'),
 ('Staffelter\xa0Hof Winery', 862, 'DEU', 'CAT9', 'Distillers, Vintners, & Breweries'),
 ('Monnaie de Paris\xa0', 864, 'FRA', 'CAT12', 'Manufacturing & Production'),
 ('The Royal Mint', 886, 'GBR', 'CAT12', 'Manufacturing & Production'),
 ("Sean's Bar", 900, 'IRL', 'CAT4', 'Cafés, Restaurants & Bars')]

## 5. Counting the categories
<p>With that extra detail about the oldest businesses, we can see that Kongō Gumi is a construction company. In that list of six businesses, we also see a café, a winery, and a bar. The two companies recorded as "Manufacturing and Production" are both mints. That is, they produce currency.</p>
<p>I'm curious as to what other industries constitute the oldest companies around the world, and which industries are most common.</p>

In [15]:
%%sql

-- Select the category and count of category (as "n")
-- arranged by descending count, limited to 10 most common categories
select category, count(*) as n
from categories
group by category
order by n desc
limit 10;

 * postgresql:///oldestbusinesses
10 rows affected.


[('Media', 1),
 ('Telecommunications', 1),
 ('Distillers, Vintners, & Breweries', 1),
 ('Defense', 1),
 ('Agriculture', 1),
 ('Cafés, Restaurants & Bars', 1),
 ('Construction', 1),
 ('Retail', 1),
 ('Energy', 1),
 ('Conglomerate', 1)]

## 6. Oldest business by continent
<p>It looks like "Banking &amp; Finance" is the most popular category. Maybe that's where you should aim if you want to start a thousand-year business.</p>
<p>One thing we haven't looked at yet is where in the world these really old businesses are. To answer these questions, we'll need to join the <code>businesses</code> table to the <code>countries</code> table. Let's start by asking how old the oldest business is on each continent.</p>

In [17]:
%%sql
select * from countries

 * postgresql:///oldestbusinesses
195 rows affected.


[('AFG', 'Afghanistan', 'Asia'),
 ('AGO', 'Angola', 'Africa'),
 ('ALB', 'Albania', 'Europe'),
 ('AND', 'Andorra', 'Europe'),
 ('ARE', 'United Arab Emirates', 'Asia'),
 ('ARG', 'Argentina', 'South America'),
 ('ARM', 'Armenia', 'Asia'),
 ('ATG', 'Antigua and Barbuda', 'North America'),
 ('AUS', 'Australia', 'Oceania'),
 ('AUT', 'Austria', 'Europe'),
 ('AZE', 'Azerbaijan', 'Asia'),
 ('BDI', 'Burundi', 'Africa'),
 ('BEL', 'Belgium', 'Europe'),
 ('BEN', 'Benin', 'Africa'),
 ('BFA', 'Burkina Faso', 'Africa'),
 ('BGD', 'Bangladesh', 'Asia'),
 ('BGR', 'Bulgaria', 'Europe'),
 ('BHR', 'Bahrain', 'Asia'),
 ('BHS', 'Bahamas', 'North America'),
 ('BIH', 'Bosnia and Herzegovina', 'Europe'),
 ('BLR', 'Belarus', 'Europe'),
 ('BLZ', 'Belize', 'North America'),
 ('BOL', 'Bolivia, Plurinational State of', 'South America'),
 ('BRA', 'Brazil', 'South America'),
 ('BRB', 'Barbados', 'North America'),
 ('BRN', 'Brunei Darussalam', 'Asia'),
 ('BTN', 'Bhutan', 'Asia'),
 ('BWA', 'Botswana', 'Africa'),
 ('CAF', 'Central African Republic', 'Africa'),
 ('CAN', 'Canada', 'North America'),
 ('CHE', 'Switzerland', 'Europe'),
 ('CHL', 'Chile', 'South America'),
 ('CHN', 'China', 'Asia'),
 ('CIV', "Côte d'Ivoire", 'Africa'),
 ('CMR', 'Cameroon', 'Africa'),
 ('COD', 'Congo, The Democratic Republic of the', 'Africa'),
 ('COG', 'Congo', 'Africa'),
 ('COL', 'Colombia', 'South America'),
 ('COM', 'Comoros', 'Africa'),
 ('CPV', 'Cabo Verde', 'Africa'),
 ('CRI', 'Costa Rica', 'North America'),
 ('CUB', 'Cuba', 'North America'),
 ('CYP', 'Cyprus', 'Europe'),
 ('CZE', 'Czechia', 'Europe'),
 ('DEU', 'Germany', 'Europe'),
 ('DJI', 'Djibouti', 'Africa'),
 ('DMA', 'Dominica', 'North America'),
 ('DNK', 'Denmark', 'Europe'),
 ('DOM', 'Dominican Republic', 'North America'),
 ('DZA', 'Algeria', 'Africa'),
 ('ECU', 'Ecuador', 'South America'),
 ('EGY', 'Egypt', 'Africa'),
 ('ERI', 'Eritrea', 'Africa'),
 ('ESP', 'Spain', 'Europe'),
 ('EST', 'Estonia', 'Europe'),
 ('ETH', 'Ethiopia', 'Africa'),
 ('FIN', 'Finland', 'Europe'),
 ('FJI', 'Fiji', 'Oceania'),
 ('FRA', 'France', 'Europe'),
 ('FSM', 'Micronesia, Federated States of', 'Oceania'),
 ('GAB', 'Gabon', 'Africa'),
 ('GBR', 'United Kingdom', 'Europe'),
 ('GEO', 'Georgia', 'Asia'),
 ('GHA', 'Ghana', 'Africa'),
 ('GIN', 'Guinea', 'Africa'),
 ('GMB', 'Gambia', 'Africa'),
 ('GNB', 'Guinea-Bissau', 'Africa'),
 ('GNQ', 'Equatorial Guinea', 'Africa'),
 ('GRC', 'Greece', 'Europe'),
 ('GRD', 'Grenada', 'North America'),
 ('GTM', 'Guatemala', 'North America'),
 ('GUY', 'Guyana', 'South America'),
 ('HND', 'Honduras', 'North America'),
 ('HRV', 'Croatia', 'Europe'),
 ('HTI', 'Haiti', 'North America'),
 ('HUN', 'Hungary', 'Europe'),
 ('IDN', 'Indonesia', 'Asia'),
 ('IND', 'India', 'Asia'),
 ('IRL', 'Ireland', 'Europe'),
 ('IRN', 'Iran, Islamic Republic of', 'Asia'),
 ('IRQ', 'Iraq', 'Asia'),
 ('ISL', 'Iceland', 'Europe'),
 ('ISR', 'Israel', 'Asia'),
 ('ITA', 'Italy', 'Europe'),
 ('JAM', 'Jamaica', 'North America'),
 ('JOR', 'Jordan', 'Asia'),
 ('JPN', 'Japan', 'Asia'),
 ('KAZ', 'Kazakhstan', 'Asia'),
 ('KEN', 'Kenya', 'Africa'),
 ('KGZ', 'Kyrgyzstan', 'Asia'),
 ('KHM', 'Cambodia', 'Asia'),
 ('KIR', 'Kiribati', 'Oceania'),
 ('KNA', 'Saint Kitts and Nevis', 'North America'),
 ('KOR', 'Korea, Republic of', 'Asia'),
 ('KWT', 'Kuwait', 'Asia'),
 ('LAO', "Lao People's Democratic Republic", 'Asia'),
 ('LBN', 'Lebanon', 'Asia'),
 ('LBR', 'Liberia', 'Africa'),
 ('LBY', 'Libya', 'Africa'),
 ('LCA', 'Saint Lucia', 'North America'),
 ('LIE', 'Liechtenstein', 'Europe'),
 ('LKA', 'Sri Lanka', 'Asia'),
 ('LSO', 'Lesotho', 'Africa'),
 ('LTU', 'Lithuania', 'Europe'),
 ('LUX', 'Luxembourg', 'Europe'),
 ('LVA', 'Latvia', 'Europe'),
 ('MAR', 'Morocco', 'Africa'),
 ('MCO', 'Monaco', 'Europe'),
 ('MDA', 'Moldova, Republic of', 'Europe'),
 ('MDG', 'Madagascar', 'Africa'),
 ('MDV', 'Maldives', 'Asia'),
 ('MEX', 'Mexico', 'North America'),
 ('MHL', 'Marshall Islands', 'Oceania'),
 ('MKD', 'North Macedonia', 'Europe'),
 ('MLI', 'Mali', 'Africa'),
 ('MLT', 'Malta',

In [22]:
%%sql

-- Select the oldest founding year (as "oldest") from businesses, 
-- and continent from countries
-- for each continent, ordered from oldest to newest 
select min(year_founded) as oldest, c.continent
from businesses as b
join countries as c
on b.country_code = c.country_code
GROUP BY c.continent, b.year_founded 
ORDER BY b.year_founded


 * postgresql:///oldestbusinesses
150 rows affected.


[(578, 'Asia'),
 (803, 'Europe'),
 (862, 'Europe'),
 (864, 'Europe'),
 (886, 'Europe'),
 (900, 'Europe'),
 (1040, 'Europe'),
 (1074, 'Europe'),
 (1135, 'Europe'),
 (1153, 'Asia'),
 (1218, 'Europe'),
 (1230, 'Europe'),
 (1248, 'Europe'),
 (1250, 'Europe'),
 (1328, 'Europe'),
 (1340, 'Europe'),
 (1346, 'Europe'),
 (1348, 'Europe'),
 (1422, 'Europe'),
 (1467, 'Europe'),
 (1511, 'Europe'),
 (1520, 'Europe'),
 (1534, 'North America'),
 (1565, 'South America'),
 (1584, 'Asia'),
 (1590, 'Europe'),
 (1606, 'Europe'),
 (1621, 'South America'),
 (1638, 'North America'),
 (1647, 'Europe'),
 (1649, 'Europe'),
 (1660, 'South America'),
 (1665, 'Europe'),
 (1670, 'North America'),
 (1694, 'South America'),
 (1703, 'North America'),
 (1721, 'Europe'),
 (1729, 'Europe'),
 (1736, 'Asia'),
 (1756, 'Europe'),
 (1770, 'North America'),
 (1772, 'Africa'),
 (1776, 'Europe'),
 (1790, 'Europe'),
 (1800, 'Asia'),
 (1808, 'Asia'),
 (1809, 'Oceania'),
 (1811, 'South America'),
 (1814, 'Africa'),
 (1819, 'Asia'),
 (1820, 'Africa'),
 (1820, 'Asia'),
 (1822, 'South America'),
 (1830, 'Asia'),
 (1830, 'North America'),
 (1835, 'Asia'),
 (1840, 'South America'),
 (1841, 'Asia'),
 (1841, 'Europe'),
 (1845, 'Asia'),
 (1847, 'Africa'),
 (1849, 'Africa'),
 (1853, 'Asia'),
 (1854, 'Africa'),
 (1858, 'Asia'),
 (1861, 'Europe'),
 (1861, 'Oceania'),
 (1862, 'North America'),
 (1864, 'Europe'),
 (1870, 'North America'),
 (1871, 'South America'),
 (1873, 'Europe'),
 (1877, 'Asia'),
 (1877, 'South America'),
 (1878, 'Africa'),
 (1878, 'Asia'),
 (1878, 'Europe'),
 (1881, 'Asia'),
 (1882, 'Europe'),
 (1883, 'Africa'),
 (1883, 'Asia'),
 (1885, 'Asia'),
 (1889, 'Africa'),
 (1890, 'Asia'),
 (1890, 'North America'),
 (1891, 'North America'),
 (1892, 'Africa'),
 (1894, 'Africa'),
 (1896, 'Africa'),
 (1897, 'Asia'),
 (1899, 'Europe'),
 (1901, 'Africa'),
 (1902, 'North America'),
 (1903, 'Asia'),
 (1904, 'Africa'),
 (1904, 'North America'),
 (1906, 'Africa'),
 (1907, 'Africa'),
 (1908, 'North America'),
 (1909, 'North America'),
 (1912, 'Europe'),
 (1913, 'Africa'),
 (1913, 'Asia'),
 (1914, 'Asia'),
 (1917, 'Africa'),
 (1920, 'North America'),
 (1921, 'Africa'),
 (1922, 'Africa'),
 (1924, 'Africa'),
 (1928, 'Asia'),
 (1929, 'North America'),
 (1930, 'Asia'),
 (1930, 'Europe'),
 (1931, 'Asia'),
 (1932, 'Asia'),
 (1933, 'Africa'),
 (1937, 'Asia'),
 (1938, 'North America'),
 (1939, 'Africa'),
 (1939, 'Asia'),
 (1943, 'Africa'),
 (1945, 'Africa'),
 (1945, 'Asia'),
 (1946, 'Africa'),
 (1947, 'Africa'),
 (1947, 'Asia'),
 (1948, 'Asia'),
 (1952, 'Asia'),
 (1954, 'Asia'),
 (1955, 'Africa'),
 (1957, 'Africa'),
 (1959, 'Asia'),
 (1960, 'Africa'),
 (1962, 'Africa'),
 (1962, 'Asia'),
 (1963, 'Africa'),
 (1964, 'Africa'),
 (1965, 'Africa'),
 (1967, 'Africa'),
 (1968, 'Africa'),
 (1971, 'Africa'),
 (1973, 'Africa'),
 (1975, 'Africa'),
 (1977, 'Asia'),
 (1978, 'Africa'),
 (1981, 'Africa'),
 (1991, 'Oceania'),
 (1994, 'Africa'),
 (1996, 'Africa'),
 (1999, 'Europe')]

## 7. Joining everything for further analysis
<p>Interesting. There's a jump in time from the older businesses in Asia and Europe to the 16th Century oldest businesses in North and South America, then to the 18th and 19th Century oldest businesses in Africa and Oceania. </p>
<p>As mentioned earlier, when analyzing data it's often really helpful to have all the tables you want access to joined together into a single set of results that can be analyzed further. Here, that means we need to join all three tables.</p>

In [31]:
%%sql

--Select the business, founding year, category, country, continent
select * 
--b.business, b.year_founded, c.category, c.country, c.continent
from businesses as b
join categories as c
on b.category_code = c.category_code

 * postgresql:///oldestbusinesses
163 rows affected.


[('Hamoud Boualem', 1878, 'CAT11', 'DZA', 'CAT11', 'Food & Beverages'),
 ('Communauté Électrique du Bénin', 1968, 'CAT10', 'BEN', 'CAT10', 'Energy'),
 ('Botswana Meat Commission', 1965, 'CAT1', 'BWA', 'CAT1', 'Agriculture'),
 ('Air Burkina', 1967, 'CAT2', 'BFA', 'CAT2', 'Aviation & Transport'),
 ('Brarudi', 1955, 'CAT9', 'BDI', 'CAT9', 'Distillers, Vintners, & Breweries'),
 ('Cameroon Development Corporation', 1947, 'CAT1', 'CMR', 'CAT1', 'Agriculture'),
 ('Correios de Cabo Verde', 1849, 'CAT16', 'CPV', 'CAT16', 'Postal Service'),
 ('Banque Internationale pour la Centrafrique', 1946, 'CAT3', 'CAF', 'CAT3', 'Banking & Finance'),
 ('Cotontchad', 1971, 'CAT1', 'TCD', 'CAT1', 'Agriculture'),
 ('Central Bank of the Comoros', 1981, 'CAT3', 'COM', 'CAT3', 'Banking & Finance'),
 ('Société nationale des Chemins de fer du Congo', 1889, 'CAT2', 'COD', 'CAT2', 'Aviation & Transport'),
 ('Development Bank of the Central African States', 1975, 'CAT3', 'COG', 'CAT3', 'Banking & Finance'),
 ('La Poste', 1945, 'CAT16', 'CIV', 'CAT16', 'Postal Service'),
 ('Ethio-Djibouti Railways', 1901, 'CAT2', 'DJI', 'CAT2', 'Aviation & Transport'),
 ('Egyptian National Railways', 1854, 'CAT2', 'EGY', 'CAT2', 'Aviation & Transport'),
 ('Guinea Ecuatorial Airlines', 1996, 'CAT2', 'GNQ', 'CAT2', 'Aviation & Transport'),
 ('Asmara Brewery', 1939, 'CAT9', 'ERI', 'CAT9', 'Distillers, Vintners, & Breweries'),
 ('National Bank of Ethiopia', 1906, 'CAT3', 'ETH', 'CAT3', 'Banking & Finance'),
 ('BGFIBank Group', 1971, 'CAT3', 'GAB', 'CAT3', 'Banking & Finance'),
 ('Halco Mining', 1962, 'CAT15', 'GIN', 'CAT15', 'Mining'),
 ('Correios da Guiné-Bissau', 1973, 'CAT16', 'GNB', 'CAT16', 'Postal Service'),
 ('KCB Group Limited', 1896, 'CAT3', 'KEN', 'CAT3', 'Banking & Finance'),
 ('Central Bank of Lesotho', 1978, 'CAT3', 'LSO', 'CAT3', 'Banking & Finance'),
 ('National Port Authority', 1921, 'CAT2', 'LBR', 'CAT2', 'Aviation & Transport'),
 ('Umma Bank', 1907, 'CAT3', 'LBY', 'CAT3', 'Banking & Finance'),
 ('Air Madagascar', 1962, 'CAT2', 'MDG', 'CAT2', 'Aviation & Transport'),
 ('Malawi Broadcasting Corporation', 1964, 'CAT13', 'MWI', 'CAT13', 'Media'),
 ('Office de Radiodiffusion-Télévision du Mali', 1957, 'CAT13', 'MLI', 'CAT13', 'Media'),
 ('Central Bank of Mauritania', 1973, 'CAT3', 'MRT', 'CAT3', 'Banking & Finance'),
 ('Mauritius Post', 1772, 'CAT16', 'MUS', 'CAT16', 'Postal Service'),
 ('Attijariwafa Bank', 1904, 'CAT3', 'MAR', 'CAT3', 'Banking & Finance'),
 ('Beira Railroad Corporation', 1892, 'CAT2', 'MOZ', 'CAT2', 'Aviation & Transport'),
 ('NamPost', 1814, 'CAT16', 'NAM', 'CAT16', 'Postal Service'),
 ('Office of Radio and Television of Niger', 1960, 'CAT13', 'NER', 'CAT13', 'Media'),
 ('First Bank of Nigeria', 1894, 'CAT3', 'NGA', 'CAT3', 'Banking & Finance'),
 ('National Post Office', 1922, 'CAT16', 'RWA', 'CAT16', 'Postal Service'),
 ('Central Bank of São Tomé and Príncipe', 1975, 'CAT3', 'STP', 'CAT3', 'Banking & Finance'),
 ('Dakar–Niger Railway', 1924, 'CAT2', 'SEN', 'CAT2', 'Aviation & Transport'),
 ('Air Seychelles', 1977, 'CAT2', 'SYC', 'CAT2', 'Aviation & Transport'),
 ('Rokel Commercial Bank', 1917, 'CAT3', 'SLE', 'CAT3', 'Banking & Finance'),
 ('Radio Mogadishu', 1943, 'CAT13', 'SOM', 'CAT13', 'Media'),
 ('Premier FMCG', 1820, 'CAT12', 'ZAF', 'CAT12', 'Manufacturing & Production'),
 ('Ivory Bank', 1994, 'CAT3', 'SSD', 'CAT3', 'Banking & Finance'),
 ('Bank of Khartoum', 1913, 'CAT3', 'SDN', 'CAT3', 'Banking & Finance'),
 ('Swazi Rail', 1963, 'CAT2', 'SWZ', 'CAT2', 'Aviation & Transport'),
 ('Tanzania Breweries Limited', 1933, 'CAT9', 'TZA', 'CAT9', 'Distillers, Vintners, & Breweries'),
 ('La Poste du Togo', 1883, 'CAT16', 'TGO', 'CAT16', 'Postal Service'),
 ('La Poste Tunisienne', 1847, 'CAT16', 'TUN', 'CAT16', 'Postal Service'),
 ('Stanbic Bank Uganda Limited', 1906, 'CAT3', 'UGA', 'CAT3', 'Banking & Finance'),
 ('ZamPost', 1896, 'CAT16', 'ZMB', 'CAT16', 'Postal Service'),
 ('Standard Chartered Zimbabwe', 1892, 'CAT3', 'ZWE', 'CAT3', 'Banking & Finance'

## 8. Counting categories by continent
<p>Having <code>businesses</code> joined to <code>categories</code> and <code>countries</code> together means we can ask questions about both these things together. For example, which are the most common categories for the oldest businesses on each continent?</p>

In [40]:
%%sql

-- Count the number of businesses in each continent and category
select country.continent, count(*)
from businesses as b
join categories as c
on b.category_code = c.category_code
join countries as country
on b.country_code = country.country_code
group by country.continent

 * postgresql:///oldestbusinesses
6 rows affected.


[('Europe', 42),
 ('Oceania', 3),
 ('Asia', 42),
 ('Africa', 50),
 ('North America', 17),
 ('South America', 9)]

## 9. Filtering counts by continent and category
<p>Combining continent and business category led to a lot of results. It's difficult to see what is important. To trim this down to a manageable size, let's restrict the results to only continent/category pairs with a high count.</p>

In [44]:
%%sql

-- Repeat that previous query, filtering for results having a count greater than 5
select country.continent, count(*) as C
from businesses as b
join categories as c
on b.category_code = c.category_code
join countries as country
on b.country_code = country.country_code
group by country.continent
having  C > 5
order by C desc

 * postgresql:///oldestbusinesses
(psycopg2.errors.UndefinedFunction) operator does not exist: categories > integer
LINE 9: having  C > 5
                  ^
HINT:  No operator matches the given name and argument type(s). You might need to add explicit type casts.

[SQL: -- Repeat that previous query, filtering for results having a count greater than 5
select country.continent, count(*) as C
from businesses as b
join categories as c
on b.category_code = c.category_code
join countries as country
on b.country_code = country.country_code
group by country.continent
having  C > 5
order by C]
(Background on this error at: http://sqlalche.me/e/13/f405)
